### DISCUSSION TITLE
Relation between runtime version on server and installed packages via NuGet

### DISCUSSION THREAD

**Discussion by Bykiev** [dotnet/runtime#93931] [General]
Hi, is it fine if in my app I'm using the higher versions of ASP .NET libs than installed on the server .NET runtime? For ex., in my app I have installed via NuGet asp net core libraries 6.0.23, but on the server (Windows server 2012r2) I've .NET runtime 6.0.16. 

I've migrated my app from .NET Framework WebAPI on .NET 6 WebAPI which is hosted on IIS. I'm experiencing strange issue: the memory usage in task manager is high enough and today I faced with such issue: the memory is grows and freed in a loop (seems GC is running), then AspNetCoreModule v2 crashed and app restarted. It happened after some time of app publish, but I don't know is it related to my issue. Can it be related with old version of .NET runtime and hosting bundle installed? I've checked the changelog and there is no changes, except security patches. I've checked my code and didn't find any places with memory leaks, all resources closed properly. 
The app itself is quite common WebAPI: I've Quartz.NET, which runs background tasks with schedule, standard controllers, DI, SignalR, MassTransit (to communicate with other services via RabbitMQ), WCF clients to get external data. The error code: w3wp.exe crash (Exception code: 0xc0000005)

**Comment by huoyaoyuan** [comment]
> Hi, is it fine if in my app I'm using the higher versions of ASP .NET libs than installed on the server .NET runtime? For ex., in my app I have installed via NuGet asp net core libraries 6.0.23, but on the server (Windows server 2012r2) I've .NET runtime 6.0.16.

There's no direct relationship between patch versions of inbox libraries from runtime and libraries from NuGet. The relationship is specified by target framework of NuGet libraries. You can install 7.0.x or 8.0.x libraries if they support net6.0, and run on .NET runtime 6.0.x.

> I've migrated my app from .NET Framework WebAPI on .NET 6 WebAPI which is hosted on IIS. I'm experiencing strange issue: the memory usage in task manager is high enough and today I faced with such issue: the memory is grows and freed in a loop (seems GC is running), then AspNetCoreModule v2 crashed and app restarted. It happened after some time of app publish, but I don't know is it related to my issue. Can it be related with old version of .NET runtime and hosting bundle installed?

Memory leak if any, can be caused by different reasons. It can be a runtime bug, or some library holding more and more objects in static field.

> The error code: w3wp.exe crash (Exception code: 0xc0000005)

0xc0000005 is access violation, which can typically be caused by null references/pointers.
